src/watchdog/observers/fsevents.py

```diff
@@ -67,86 +71,125 @@ class FSEventsEmitter(EventEmitter):
+                        self.queue_event(sub_event)
+                    i += 1
+                elif os.path.exists(event.path):
+                    cls = DirCreatedEvent if event.is_directory else FileCreatedEvent
+                    self.queue_event(cls(src_path))
+                    self.queue_event(DirModifiedEvent(os.path.dirname(src_path)))
+                else:
+                    cls = DirDeletedEvent if event.is_directory else FileDeletedEvent
+                    self.queue_event(cls(src_path))
+                    self.queue_event(DirModifiedEvent(os.path.dirname(src_path)))
 
-    def queue_events(self, timeout):
-        with self._lock:
-            events = self.native_events
-            i = 0
-            while i < len(events):
-                event = events[i]
-                src_path = self._encode_path(event.path)
-
-                # For some reason the create and remove flags are sometimes also
-                # set for rename and modify type events, so let those take
-                # precedence.
-                if event.is_renamed:
-                    # Internal moves appears to always be consecutive in the same
-                    # buffer and have IDs differ by exactly one (while others
-                    # don't) making it possible to pair up the two events coming
-                    # from a singe move operation. (None of this is documented!)
-                    # Otherwise, guess whether file was moved in or out.
-                    # TODO: handle id wrapping
-                    if (i + 1 < len(events) and events[i + 1].is_renamed
-                            and events[i + 1].event_id == event.event_id + 1):
-                        cls = DirMovedEvent if event.is_directory else FileMovedEvent
-                        dst_path = self._encode_path(events[i + 1].path)
-                        self.queue_event(cls(src_path, dst_path))
-                        self.queue_event(DirModifiedEvent(os.path.dirname(src_path)))
-                        self.queue_event(DirModifiedEvent(os.path.dirname(dst_path)))
-                        i += 1
-                    elif os.path.exists(event.path):
-                        cls = DirCreatedEvent if event.is_directory else FileCreatedEvent
-                        self.queue_event(cls(src_path))
-                        self.queue_event(DirModifiedEvent(os.path.dirname(src_path)))
-                    else:
-                        cls = DirDeletedEvent if event.is_directory else FileDeletedEvent
+            if event.is_created:
+                cls = DirCreatedEvent if event.is_directory else FileCreatedEvent
+                if not event.is_coalesced or (
+                    event.is_coalesced and not event.is_renamed and not event.is_modified and not
+                    event.is_inode_meta_mod and not event.is_xattr_mod
+                ):
+                    if src_path not in self.filesystem_view:
```

Comment:
There are some cases where `src_path not in self.filesystem_view` might evaluate to False even though the item was actually newly created. This will be the case when we missed a previous deleted event or moved event which should have removed `src_path` from `filesystem_view`, for instance when the user changes just the casing of a filename (see previous comment). There may be other corner cases which I am not aware of and which will require careful workarounds.